### PR TITLE
feat(globaloffensive): add types for globaloffensive

### PR DESCRIPTION
**Pull request by joshuajeschek**
https://github.com/DoctorMcKay/node-globaloffensive

some attributes have been intentionally set as `unknown | null`, since I wasn't able to test their values.

In my case, the values were always null.
 Since this package relies on being logged in as a steam user and making requests, some values would require special events to be filled at all (e.g. being in a match (which isn't possible since you cannot run the game while being logged in with node-globaloffensive) or a big csgo tournament happening or even being a part of a tournament team)

if a type is unknown[], an empty array was all I could get from usage.

minimum TypeScript Version set because of steam-user.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/globaloffensive/enums.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export enum GCConnectionStatus {
+    HAVE_SESSION = 0,
+	GC_GOING_DOWN = 1,
```
Comment (DangerBotOSS) [inline types/globaloffensive/enums.d.ts:3]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/globaloffensive/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+// Project: https://github.com/DoctorMcKay/node-globaloffensive
+// Definitions by: joshuajeschek <https://github.com/joshuajeschek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.2
+
+// minimum TypeScript Version set because of steam-user.
+// some attributes have been intentionally set as `unknown | null`, since I wasn't able to test their values.
+// In my case, the values were always null.
+// if a type is unknown[], an empty array was all I could get from usage.
+
```
Comment (peterblazejewicz) [inline types/globaloffensive/index.d.ts:11]: Can you move this comment into PR message, thx!

**Comment by peterblazejewicz** [comment]
aren't the `enums` constant symbols? so they can be used that way, e.g.:
```ts
export const GCConnectionStatus: {
    HAVE_SESSION: 0,
    GC_GOING_DOWN: 1,
    NO_SESSION: 2,
    NO_SESSION_IN_LOGON_QUEUE: 3,
    NO_STEAM: 4
}
```
```ts
static GCConnectionStatus: typeof GCConnectionStatus;
```
```ts
type ItemCustomizationNotificationType =
    typeof ItemCustomizationNotification[keyof typeof ItemCustomizationNotification];
```
```ts
    itemCustomizationNotification: [itemIds: string[], notificationType: ItemCustomizationNotificationType];
```

test (from readme):
```ts
csgo.on('itemCustomizationNotification', (itemIds, notificationType) => {
    if (notificationType == GlobalOffensive.ItemCustomizationNotification.CasketInvFull) {
        console.log('Storage unit ' + itemIds[0] + ' is full');
    }
});
```

**Comment by peterblazejewicz** [comment]
please run prettier on this package as per readme:
```ts
npm run prettier -- --write types/globaloffensive/**/*.ts
```

**Comment by joshuajeschek** [comment]
Thanks for the feedback! I switched to const, and defined ValueOf<T>, because like this it is still possible to see the original object when for example hovering over the `notificationType` parameter in IDE. By just doing `typeof ItemCustomizationNotification[keyof typeof ItemCustomizationNotification]` I would just see the numbers, which I found to be a bit weird 🤔

**Review by peterblazejewicz — APPROVED** [review]
@joshuajeschek thx!
LGTM!

**Comment by joshuajeschek** [comment]
Ready to merge, thanks as always for your review!

**Pull request merged**